src/components/FavoriteButton/FavoriteButton.jsx

```diff
@@ -0,0 +1,37 @@
+import React, { useEffect, useState } from 'react';
+import { FaStar } from 'react-icons/fa';
+import useDetail from '../../hooks/useDetail';
+import useGlobal from '../../hooks/useGlobal';
+import { Button } from './styled';
+
+export default function FavoriteButton({ video }) {
+  const {
+      globalState,
+      videoIsInFavorites,
+      addToFavorites,
+      removeFromFavorites,
+    } = useGlobal(),
+    { detailState } = useDetail(),
+    [buttonMessage, setButtonMessage] = useState(''),
+    handleClick = () => {
+      if (videoIsInFavorites(video)) {
+        removeFromFavorites(video);
+      } else {
+        addToFavorites(video);
+      }
+    };
```

Comment:
It is not recommended to skip the reserved word `const`. It is not readable, I recommend having all the necessary `const` by line.